core/services/fluxmonitorv2/flux_monitor_test.go

```diff
@@ -0,0 +1,1674 @@
+	)
+	spec.PollTimerDisabled = true
+	spec.IdleTimerDisabled = true
+
+	const fetchedAnswer = 100
+	answerBigInt := big.NewInt(fetchedAnswer * int64(math.Pow10(int(spec.Precision))))
+
+	chRoundState1 := make(chan struct{})
+	chRoundState2 := make(chan struct{})
+
+	logBroadcaster.On("Register", mock.Anything, mock.Anything).Return(true)
+	logBroadcaster.On("Unregister", mock.Anything, mock.Anything)
+
+	fluxAggregator.On("Address").Return(spec.ContractAddress).Maybe()
+	fluxAggregator.On("GetOracles", nilOpts).Return(oracles, nil)
+	fluxAggregator.On("LatestRoundData", nilOpts).Return(makeRoundDataForRoundID(1), nil).Once()
+	fluxAggregator.On("OracleRoundState", nilOpts, nodeAddr, uint32(1)).Return(flux_aggregator_wrapper.OracleRoundState{
+		RoundId:          1,
+		EligibleToSubmit: false,
+		LatestSubmission: answerBigInt,
+		StartedAt:        now(),
+		Timeout:          uint64(1000000),
+	}, nil).Once()
+
+	startedAt := uint64(time.Now().Unix())
+	timeout := uint64(3)
+	fluxAggregator.On("OracleRoundState", nilOpts, nodeAddr, uint32(0)).Return(flux_aggregator_wrapper.OracleRoundState{
+		RoundId:          1,
+		EligibleToSubmit: false,
+		LatestSubmission: answerBigInt,
+		StartedAt:        startedAt,
+		Timeout:          timeout,
+	}, nil).Once().
+		Run(func(mock.Arguments) { close(chRoundState1) }).
+		Once()
+	fluxAggregator.On("OracleRoundState", nilOpts, nodeAddr, uint32(0)).Return(flux_aggregator_wrapper.OracleRoundState{
+		RoundId:          1,
+		EligibleToSubmit: false,
+		LatestSubmission: answerBigInt,
+		StartedAt:        startedAt,
+		Timeout:          timeout,
+	}, nil).Once().
+		Run(func(mock.Arguments) { close(chRoundState2) }).
+		Once()
+
+	fm, err := fluxmonitorv2.NewFluxMonitor(
+		NewPipelineRun(),
+		fluxmonitorv2.NewStore(store.DB, store, jobORM, pipelineORM),
+		pollTicker,
+		fluxmonitorv2.NewPaymentChecker(assets.NewLink(1), nil),
+		spec.ContractAddress,
+		fluxmonitorv2.NewFluxAggregatorContractSubmitter(fluxAggregator, transmissionAddress),
+		fluxmonitorv2.NewDeviationChecker(spec.Threshold, spec.AbsoluteThreshold),
+		fluxmonitorv2.Flags{},
+		fluxAggregator,
+		logBroadcaster,
+		spec,
+		func() {},
+		big.NewInt(0),
+		big.NewInt(100000000000),
+	)
+	require.NoError(t, err)
+
+	fm.Start()
+	fm.OnConnect()
+
+	logBroadcast := new(logmocks.Broadcast)
+	logBroadcast.On("WasAlreadyConsumed").Return(false, nil)
+	logBroadcast.On("DecodedLog").Return(&flux_aggregator_wrapper.FluxAggregatorNewRound{RoundId: big.NewInt(0), StartedAt: big.NewInt(time.Now().UTC().Unix())})
+	logBroadcast.On("MarkConsumed").Return(nil)
+	fm.HandleLog(logBroadcast, nil)
+
+	gomega.NewGomegaWithT(t).Eventually(chRoundState1).Should(gomega.BeClosed())
+	gomega.NewGomegaWithT(t).Eventually(chRoundState2).Should(gomega.BeClosed())
+
+	time.Sleep(time.Duration(2*timeout) * time.Second)
+	fm.Close()
+
+	fluxAggregator.AssertExpectations(t)
+}
+
+func TestFluxMonitor_SufficientFunds(t *testing.T) {
```

Comment:
This now delegates to the `PaymentChecker` but I've left this in for sanity checks for now.

Reply:
I now call the `PaymentChecker.SufficientFunds` directly and have removed this test